enhancements/local-storage/subprovisioner-integration-into-lvms.md

```diff
@@ -0,0 +1,361 @@
+
+### Test Plan
+
+- **Integration Tests**:
+  - Update existing LVMS integration tests to include scenarios for shared storage provisioning with Subprovisioner.
+  - Ensure that device detection and VG management are functioning correctly with both TopoLVM and Subprovisioner.
+  - QE will be extending the existing test suites to include shared storage provisioning and synchronization tests.
+
+- **E2E Tests**:
+  - Implement end-to-end tests to validate the complete workflow from device discovery to VG provisioning and usage.
+  - Include multi-node scenarios to test shared storage provisioning and synchronization.
+
+- **Performance and Stress Tests**:
+  - Conduct performance tests to assess the scalability and robustness of the VG management and CSI driver operations.
+  - The performance tests will have the same scope as the existing TopoLVM performance tests, mainly provisioning times and I/O
+  - Perform stress tests to evaluate system behavior under high load and failure conditions.
+  - We will run these tests before any graduation to GA at the minimum.
+
+### Graduation Criteria
+
+- **Developer Preview (Early Evaluation and Feedback)**:
+  - Initial implementation with basic functionality for shared and node-local VG provisioning.
+  - Basic integration and E2E tests in place.
+  - Feedback from early adopters and stakeholders collected.
+  - No official Product Support.
+  - Functionality is provided with very limited, if any, documentation. Documentation is not included as part of the product’s documentation set.
+
+- **Technology Preview**:
+  - Feature-complete implementation with all planned functionality.
+  - Comprehensive test coverage including performance and stress tests.
+  - Functionality is documented as part of the products documentation set (on the Red Hat Customer Portal) and/or via the release notes.
+  - Functionality is provided with LIMITED support by Red Hat. Customers can open support cases, file bugs, and request feature enhancements. However, support is provided with NO commercial SLA and no commitment to implement any changes.
+  - Functionality has undergone more complete Red Hat testing for the configurations supported by the underlying product.
+  - Functionality is, with rare exceptions, on Red Hat’s product roadmap for a future release.
+
+- **GA**:
+  - Proven stability and performance in production-like environments.
+  - Positive feedback from initial users.
+  - Full documentation, including troubleshooting guides and best practices.
+  - Full LVMS Support Lifecycle
+
+### Upgrade / Downgrade Strategy
+
+- **Upgrade**:
+  - Ensure that upgrades are seamless with no downtime for existing workloads. Migrating to a subprovisioner enabled version is a no-break operation
+  - Test upgrade paths thoroughly to ensure compatibility and data integrity. The subprovisioner to topolvm (or vice versa) switch should be excluded and forbidden explicitly.
```

Comment:
[Workflow 6](https://github.com/openshift/enhancements/pull/1620/files#diff-5938e75542daf7b772b668e7b4d40ffa2fea036528bdaa76409254c1ca2bdd20R96) mentions that users can provision TopoLVM and shared storage side-by-side. Does that mean that a user can start with TopoLVM and add Subprovisioner but they cannot start with TopoLVM and then switch to Subprovisioner? If so, I think I understand the case; users can start with a _root_ storage driver and can add a storage driver, but they cannot change the _root_ storage driver.

Reply:
I think this is correct and a good implication that I will add. The "root" storage driver, is called the default in LVMS terminology, and there can be at most one at least none. The default driver is used to indicate to Kubernetes if we should indicate a generated storage class as default storage class.

If someone starts with TopoLVM / Subprovisioner as default storageClass, then they cannot change that without tearing down the LVMCluster.